web/source/osk/preProcessor.ts

```diff
@@ -98,10 +98,38 @@ namespace com.keyman.osk {
           Lkc.source = touch;
           Lkc.keyDistribution = keyDistribution;
         }
-        return keyman.textProcessor.processKeyEvent(Lkc, e);
+
+        if(!keyman.isEmbedded) {
+          keyman.uiManager.setActivatingUI(true);
+          com.keyman.dom.DOMEventHandlers.states._IgnoreNextSelChange = 100;
+          keyman.domManager.focusLastActiveElement();
+          com.keyman.dom.DOMEventHandlers.states._IgnoreNextSelChange = 0;
+        }
+
+        let retVal = PreProcessor.handleClick(Lkc, e);
+
+        // Now that processing is done, we can do a bit of post-processing, too.
+        keyman.uiManager.setActivatingUI(false);	// I2498 - KeymanWeb OSK does not accept clicks in FF when using automatic UI
+        return retVal;
       } else {
         return true;
       }
     }
+
+    // Created during refactoring for web-core.  Mostly serves to hold DOM-dependent
```

Comment:
This comment won't actually mean much in a year or two. Is this a temporary function which will be deleted after refactoring (if so, note that); otherwise, can you make the comment explain the purpose of the function to an outside reader?